pandas/core/indexes/multi.py

```diff
@@ -946,7 +946,7 @@ def _format_native_types(self, na_rep='nan', **kwargs):
             new_codes.append(level_codes)
 
         if len(new_levels) == 1:
```

Comment:
add a comment here that this is a single-level MI

Reply:
what do new_levels and new_codes look like here (from your test)

Reply:
The value of `new_levels` is `[array(['1', '2', '3'], dtype='<U21')]`.
The value of `new_codes` is `[FrozenNDArray([0, 2], dtype='int8')]`

Reply:
Added a comment in d47089ee3ef92152f9577602b84073455bb49763

Reply:
ok use levels[0].take(codes[0])

Reply:
Done in 159a828e910d1d2f2b17abcd7a7fb0b5acc79b7a